### PR TITLE
Handle non ascii

### PR DESCRIPTION
**Pull request by alex-silverman**

### CODE REVIEW

File: public-resources/EvaChecks.csv [hunk 1]
```diff
@@ -1,123 +1,124 @@
...
+file structure,High Priority,You uploaded an unhashed data set,"Export.csv - HashStatus, Client.csv - FirstName","You have uploaded an unhashed version of the HMIS CSV Export. If you are not sure how to run the hashed HMIS CSV Export in your  HMIS, please contact your HMIS vendor.",,126
+file structure,High Priority,Your HMIS CSV Export is out of date,Export.csv - CSVVersion column,"It looks like you either uploaded an FY2022 HMIS CSV Export or your vendor needs to update your Export.csv's CSV Version. If you are not sure how to obtain an FY2024 HMIS CSV Export in your HMIS, please contact your HMIS vendor.",,124
+file structure,High Priority,Your zip file is misstructured,Determined from file metadata,It looks like you may have unzipped your HMIS csv because the individual csv files are contained within a subdirectory.,,122
+file structure,High Priority,Your zip file contains non-ASCII characters,Specific data elements will be identified in the Detail column of a CoC's Integrity Checks export.,Contact your vendor to ensure non-ASCII characters are excluded from your export.,,134
```
Comment (alex-silverman) [inline public-resources/EvaChecks.csv:18]: Change "Your zip file contains non-ASCII characters" to "Non-ASCII Characters"

File: 03_file_structure_analysis.R [hunk 2]
```diff
@@ -15,6 +15,51 @@ high_priority_columns <- cols_and_data_types %>%
...
+    non_ascii_data <- get(file) %>%
+      mutate_all(~ifelse(!stri_enc_isascii(.), ., NA))
+    
+    # Find rows that contain any non-ASCII characters
+    non_ascii_rows <- apply(non_ascii_data, 1, function(x) any(!is.na(x)))
+    
+    if(any(non_ascii_rows)) {
+      non_ascii_info <- data.frame(
+        Column = names(non_ascii_data)[which(!is.na(non_ascii_data[non_ascii_rows, ]))],
+        Detail = paste0(
```
Comment (alex-silverman) [inline 03_file_structure_analysis.R:35]: Detail should include CSV name

File: 03_file_structure_analysis.R [hunk 3]
```diff
@@ -15,6 +15,51 @@ high_priority_columns <- cols_and_data_types %>%
...
 
+
+non_ascii_files <- function() {
+  library(stringi)
+  # Initialize an empty dataframe to store the results
+  files_with_non_ascii <- data.frame(Column = character(), Detail = character())
+  
+  for(file in unique(cols_and_data_types$File)) {
+    # Replace ASCII characters with NA
+    non_ascii_data <- get(file) %>%
```
[the remark this answers is not in the document]
Comment (alex-silverman) [reply 03_file_structure_analysis.R:26]: Also need to add "File" as a field so we can merge with cols_and_data_types

File: 03_file_structure_analysis.R [hunk 4]
```diff
@@ -15,6 +15,51 @@ high_priority_columns <- cols_and_data_types %>%
...
+  }
+  
+  return(files_with_non_ascii)
+}
+
+# non-ascii --------------------------------------------------------------
+files_with_non_ascii <- non_ascii_files() %>%
+  left_join(cols_and_data_types %>% 
+              select(Column, DataTypeHighPriority),
+            by = "Column") %>%
```
Comment (alex-silverman) [inline 03_file_structure_analysis.R:59]: should merge on Column AND File so it's unique

File: 03_file_structure_analysis.R [hunk 5]
```diff
@@ -23,8 +23,16 @@ non_ascii_files <- function() {
...
-      mutate_all(~ifelse(!stri_enc_isascii(.) | grepl("[\\(\\)\\[\\]\\<\\>\\{\\}]", .), ., NA))
+      mutate_all( ~ ifelse(
+        !stri_enc_isascii(.) |
+          str_detect(., ".\\(|\\)|\\[|\\]|\\<|\\>|\\{|\\}"),
+        .,
+        NA
+      ))
 
+    if (file == "Project") {
+      browser()
```
Comment (kiadso) [inline 03_file_structure_analysis.R:34]: be sure you remove your browser()

**Review by kiadso — APPROVED** [review]
Thanks Alex!! this is great

**Pull request merged**